### ISSUE TITLE
DIFAT chain termination

### ISSUE THREAD

**Issue by jmacadie** [mdsteele/rust-cfb#49]
I have come across an issue when using this library on an old Excel file. The  DIFAT chain does not terminate with 0xFFFFFFFE (ENDOFCHAIN) as it is meant to but goes straight to 0xFFFFFFFF (FREESECT). This then causes the library to return an error on opening the file

I have checked and Excel merrily opens the file without reporting any corruption. I assume this is a tolerated relaxation of the specification in the wild

![image](https://github.com/mdsteele/rust-cfb/assets/1902025/cf55224a-c196-454b-91e0-2e76ab506fac)

**Comment by jmacadie** [comment]
The fix is quite simple, I think. I'll submit a PR tomorrow but I'm out of time today. Just want to do a few more checks

**Comment by jmacadie** [comment]
To note I hadn't quite diagnosed the issue correctly when I first posted above. It is the _whole DIFAT page_ that must end in a ENDOFCHAIN marker. My troublesome Excel file also had a FREESECT here and this was the real reason rust-cfb was throwing an error.

I have fixed the correct issue with the submitted PR

![image](https://github.com/mdsteele/rust-cfb/assets/1902025/1f72740b-7722-4143-b66b-58d8e5d6bcf7)
